### DISCUSSION TITLE
replaceChildren VS display: none

### DISCUSSION THREAD

**Discussion by kkojae91** [woowacourse-study/2022-code-review-study#56] [Youtube] [answered]
### [#154](https://github.com/woowacourse/javascript-youtube-classroom/pull/154#discussion_r832152364) replaceChildren 예시
리뷰어: 기존 list에 skeleton으로만 replace 할 수 있을 것 같은데 한번 고민해 보세요.

replaceChildren을 통해 기존의 videoList를 skeleton으로 변경해주는 것과 
display: none 속성을 활용한 방법 중 더 좋은 방법은 무엇일까요...??!

다시 설명하면,
videoList들을 하나씩 선택해서 제거 하고 skeleton을 display: none; 해주는 것과 
videoList 전체를 replaceChildren으로 skeleton UI로 갈아 끼워주고 display: none; 을 주는 것 중 더 좋은 선택이 무엇일까요..?!

그냥... 궁금하네요! 😯

**Comment by byhhh2** [comment, marked as answer]
저는 리뷰어님의 말씀을 *elements들을 하나씩 `remove()`하는 것보다 `replaceChildren()`을 쓰는게 낫지 않나요?!* 로 이해했습니다.

둘 중 어떤게 성능이 더 좋을지는 전혀 모르겠지만,,
리뷰어님의 말씀은.. 반복문으로 하나씩 지우는 것 보다는 싹 다 지워주는 API가 있으면 걔를 쓰자! 요론..느낌 아닐까요?

그런데~~ 사실 두분이 중점을 두는 부분이 다른 것 같아요..

**크루분** : video-item만 지우고 skelton은 display로 조정하니깐 최적화다. -> **복잡하더라도 최적화에 중심을 둠**
**리뷰어** : video-item을 지우는 과정이 반복문으로 이뤄지는 데 얘는 `replaceChildren()`을 쓰는게 낫지 않나. -> 대신 skeleton이 display로 조정되는 것이 아니라 다시 dom에 끼워지는 것임. -> **복잡하지 않게 간단한 함수를 쓰자**

.. 스읍 잘 모르겠네요. 최적화냐 vs 간단하게냐 요런 문제일까요

**Reply by wonsss** [reply]
" 이 논의도 reflow , repaint 와 관련 있다" - 나그네-

**Comment by lokba** [comment]


**Reply by wonsss** [reply]
css가 항상 성능상 더 좋은가요?

**Comment by lokba** [comment]
포코가 공유해준 [Reflow와 Repaint](https://yoonucho.github.io/review/2019/11/22/reflow&repaint.html)

**Reply by wonsss** [reply]
- by 나그네 -

**Reply by lokba** [reply]
[추가로 공유한 포스팅](https://dev.to/gopal1996/understanding-reflow-and-repaint-in-the-browser-1jbg)

**Comment by wonsss** [comment]
"visibility: hidden은 display: none과 다릅니다." 
공부하신 분은 댓글좀..

**Reply by wonsss** [reply]
전자(visibility: hidden)는 요소를 보이지 않게 만들지만, 이 요소는 여전히 레이아웃에서 공간을 차지합니다(즉, 비어 있는 상자로 렌더링됨). 
반면, 후자(display: none)는 요소가 보이지 않으며 레이아웃에 포함되지도 않도록 렌더링 트리에서 요소를 완전히 제거합니다.
https://developers.google.com/web/fundamentals/performance/critical-rendering-path/render-tree-construction?hl=ko
